### ISSUE TITLE
Mojaloop Oracle Simulator is not correctly filtering results when SubId request param is provided

### ISSUE THREAD

**Issue by mdebarros** [mojaloop/project#2373]
**Summary**:
Mojaloop Oracle Simulator is not correctly filtering results when SubId request param is provided.

```bash
$ curl http://localhost:8080/oracle/participants/MSISDN/27713803912\?SubId=30
{"partyList":[{"fspId":"payeefsp","currency":"USD"},{"fspId":"payeefsp","currency":"USD","partySubIdOrType":"30"}]}
```

The above result incorrectly returns the first record `{"fspId":"payeefsp","currency":"USD"}` which dos not contain a match `partySubIdOrType=30` (unlike the 2nd result).

**Severity**:
Low

**Priority**:
Low

**Expected Behavior**

In the above example only the record with matching `partySubIdOrType=30` should be returned:

```bash
$ curl http://localhost:8080/oracle/participants/MSISDN/27713803912\?SubId=30
{"partyList":[{"fspId":"payeefsp","currency":"USD","partySubIdOrType":"30"}]}
```

**Steps to Reproduce**
  1. Ensure that there is a record created on Oracle for the same PartyType-PartyId with and without a SubId
  1. Send CURL command like the example above for that participant

**Specifications**
  - Component (if known): Mojaloop-Simulator
  - Version: v11.4.3
  - Platform: moja1
  - Subsystem: n/a
  - Type of testing: manual
  - Bug found/raised by: @mdebarros 

**Notes**:

- Severity when opened:
- Priority when opened:

**Comment by mdebarros** [comment]
PR to fix this --> https://github.com/mojaloop/simulator/pull/230

**Comment by mdebarros** [comment]
> PR to fix this --> [mojaloop/simulator#230](https://github.com/mojaloop/simulator/pull/230)

PR for [Helm v13.0.2](https://github.com/mojaloop/helm/pull/448).

```text
Test Suite:GP Tests
Environment:moja2.test.mojaloop.live
┌───────────────────────────────────────────────────┐
│                      SUMMARY                      │
├───────────────────┬───────────────────────────────┤
│ Total assertions  │ 2064                          │
├───────────────────┼───────────────────────────────┤
│ Passed assertions │ 2064                          │
├───────────────────┼───────────────────────────────┤
│ Failed assertions │ 0                             │
├───────────────────┼───────────────────────────────┤
│ Total requests    │ 446                           │
├───────────────────┼───────────────────────────────┤
│ Total test cases  │ 99                            │
├───────────────────┼───────────────────────────────┤
│ Passed percentage │ 100.00%                       │
├───────────────────┼───────────────────────────────┤
│ Started time      │ Tue, 17 Aug 2021 08:46:09 GMT │
├───────────────────┼───────────────────────────────┤
│ Completed time    │ Tue, 17 Aug 2021 08:48:32 GMT │
├───────────────────┼───────────────────────────────┤
│ Runtime duration  │ 142526 ms                     │
└───────────────────┴───────────────────────────────┘
```
[View Report](https://mojaloop-oss-qa-results.s3.us-west-2.amazonaws.com/ttk-tests/reports/TTK-Assertion-Report-multi-2021-08-17T08%3A48%3A32.170Z.html)
